utils/include/detray/simulation/landau_distribution.hpp

```diff
@@ -0,0 +1,293 @@
+            5.799512f,  5.831694f,  5.864161f,  5.896918f,  5.929968f,
+            5.963316f,  5.996967f,  6.030925f,  6.065194f,  6.099780f,
+            6.134687f,  6.169921f,  6.205486f,  6.241387f,  6.277630f,
+            6.314220f,  6.351163f,  6.388465f,  6.426130f,  6.464166f,
+            6.502578f,  6.541371f,  6.580553f,  6.620130f,  6.660109f,
+            6.700495f,  6.741297f,  6.782520f,  6.824173f,  6.866262f,
+            6.908795f,  6.951780f,  6.995225f,  7.039137f,  7.083525f,
+            7.128398f,  7.173764f,  7.219632f,  7.266011f,  7.312910f,
+            7.360339f,  7.408308f,  7.456827f,  7.505905f,  7.555554f,
+            7.605785f,  7.656608f,  7.708035f,  7.760077f,  7.812747f,
+            7.866057f,  7.920019f,  7.974647f,  8.029953f,  8.085952f,
+            8.142657f,  8.200083f,  8.258245f,  8.317158f,  8.376837f,
+            8.437300f,  8.498562f,  8.560641f,  8.623554f,  8.687319f,
+            8.751955f,  8.817481f,  8.883916f,  8.951282f,  9.019600f,
+            9.088889f,  9.159174f,  9.230477f,  9.302822f,  9.376233f,
+            9.450735f,  9.526355f,  9.603118f,  9.681054f,  9.760191f,
+            9.840558f,  9.922186f,  10.005107f, 10.089353f, 10.174959f,
+            10.261958f, 10.350389f, 10.440287f, 10.531693f, 10.624646f,
+            10.719188f, 10.815362f, 10.913214f, 11.012789f, 11.114137f,
+            11.217307f, 11.322352f, 11.429325f, 11.538283f, 11.649285f,
+            11.762390f, 11.877664f, 11.995170f, 12.114979f, 12.237161f,
+            12.361791f, 12.488946f, 12.618708f, 12.751161f, 12.886394f,
+            13.024498f, 13.165570f, 13.309711f, 13.457026f, 13.607625f,
+            13.761625f, 13.919145f, 14.080314f, 14.245263f, 14.414134f,
+            14.587072f, 14.764233f, 14.945778f, 15.131877f, 15.322712f,
+            15.518470f, 15.719353f, 15.925570f, 16.137345f, 16.354912f,
+            16.578520f, 16.808433f, 17.044929f, 17.288305f, 17.538873f,
+            17.796967f, 18.062943f, 18.337176f, 18.620068f, 18.912049f,
+            19.213574f, 19.525133f, 19.847249f, 20.180480f, 20.525429f,
+            20.882738f, 21.253102f, 21.637266f, 22.036036f, 22.450278f,
+            22.880933f, 23.329017f, 23.795634f, 24.281981f, 24.789364f,
+            25.319207f, 25.873062f, 26.452634f, 27.059789f, 27.696581f,
+            28.365274f, 29.068370f, 29.808638f, 30.589157f, 31.413354f,
+            32.285060f, 33.208568f, 34.188705f, 35.230920f, 36.341388f,
+            37.527131f, 38.796172f, 40.157721f, 41.622399f, 43.202525f,
+            44.912465f, 46.769077f, 48.792279f, 51.005773f, 53.437996f,
+            56.123356f, 59.103894f};
+
+        if (z <= 0) {
+            // Is using max OK?
+            return -std::numeric_limits<scalar_type>::max();
+        }
+        if (z >= 1) {
+            return std::numeric_limits<scalar_type>::max();
+        }
+
+        scalar_type ranlan = 0.f, u = 0.f, v = 0.f;
+        u = 1000.f * z;
+        int i = int(u);
```

Comment:
Just as an idea: Maybe you could use ```std::trunc``` here

Reply:
ok

Reply:
hmm I will just use `static_cast<int>`